docs/data/routes/docs/nextjs/data-fetching/data-services/en.md

```diff
@@ -3,4 +3,38 @@ name: data-services
 routeTemplate: ./data/component-templates/article.yml
 title: Data services used by Next.js with JSS
 ---
-# Data services used by Next.js with JSS
\ No newline at end of file
+# Data services used by Next.js with JSS
+The sample app contains `SitecorePagePropsFactory` in `src/lib/page-props-factory.ts`. This factory is responsible for fetching all needed data in order to generate `SitecorePageProps`, it exposes public method `create(context: GetServerSidePropsContext | GetStaticPropsContext)`. Read more about `context` argument [here](https://nextjs.org/docs/basic-features/data-fetching). Definition of `SitecorePageProps` you can look at [`src/lib/page-props.ts`](https://github.com/Sitecore/jss/blob/master/samples/nextjs/src/lib/page-props.ts).
+
+Currently, we are using standard REST-based dictionary and layout services. In the near future we'll also have GraphQL-based implementations available (for Sitecore Experience Edge).
+
+It doesn't matter if you are working in `SSG/SSR` you can use `sitecorePagePropsFactory.create(context)` in `getStaticProps/getServerSideProps`, `SitecorePagePropsFactory` handles all possible cases.
+
+Let's look at `SitecorePageProps` definition in details:
+```ts
+export type SitecorePageProps = {
+  locale: string;
+  layoutData: LayoutServiceData | null;
+  dictionary: DictionaryPhrases;
+  componentProps: ComponentPropsCollection;
+  notFound: boolean;
+};
+```
+### locale
+Use `context` locale if Next.js i18n is configured, otherwise use `config.language` defined in `package.json`.
+
+### layoutData
+It's `LayoutServiceData` from Sitecore Layout Service fetched by `RestLayoutService` instance. Read more about [Layout Service](/docs/fundamentals/services/layout-service). `RestLayoutService` fetch layout data using the Sitecore Layout Service REST API. Uses Axios as the default data fetcher. If you are working in `SSR` mode it will setup default required request and response `headers` using `context.req` and `context.res` attributes.
```

Comment:
> it will setup default required request and response `headers`

Maybe worth noting that this is specifically to support Sitecore tracking?

Reply:
@nickwesselman Agree, added this note.